include/net/socket.h

```diff
@@ -880,12 +880,14 @@ struct net_socket_register {
 	int (*handler)(int family, int type, int proto);
 };
 
-#define NET_SOCKET_GET_NAME(socket_name)	\
-	(__net_socket_register_##socket_name)
+#define NET_SOCKET_DEFAULT_PRIO CONFIG_NET_SOCKETS_PRIORITY_DEFAULT
 
-#define NET_SOCKET_REGISTER(socket_name, _family, _is_supported, _handler) \
+#define NET_SOCKET_GET_NAME(socket_name, prio)	\
+	(__net_socket_register_##prio##_##socket_name)
```

Comment:
So the priority is just part of the variable name, will this always make sure that the sockets are processed in this order?

Reply:
Yes, the items within an iterable section are sorted by name (https://github.com/zephyrproject-rtos/zephyr/blob/main/include/linker/linker-defs.h#L67). Similar mechanism is used in `SYS_INIT`, where the priority is just concatenated with the section name.